ext/js/pages/settings/dictionary-controller.js

```diff
@@ -1139,9 +1139,25 @@ export class DictionaryController {
         downloadUrl = downloadUrl ?? dictionaryInfo.downloadUrl;
         if (typeof downloadUrl !== 'string') { throw new Error('Attempted to update dictionary without download URL'); }
 
-        await this._deleteDictionary(dictionaryTitle);
+        const options = await this._settingsController.getOptionsFull();
+        const {profiles} = options;
+
+        /** @type {import('settings-controller.js').ProfilesDictionarySettings} */
+        const profilesDictionarySettings = {};
 
-        this._settingsController.trigger('importDictionaryFromUrl', {url: downloadUrl});
+        for (let i = 0, ii = profiles.length; i < ii; ++i) {
```

Comment:
why do we need `ii` here instead of just doing `i < profiles.length`?

Reply:
i think we can probably just use `for... of` syntax here since the index isn't used elsewhere

Reply:
the question should be why the codebase has for loop written in that way

Reply:
we dont have to follow it 😤